### PR TITLE
Remove BlockchainBuilder, speed up ChainHandler test that processes m…

### PR DESCRIPTION
**Pull request by Christewart**
…ainnet headers, add test for ChainAppConfig

The goal of this PR is to improve chain projects. This also removes `BlockchainBuilder` as we are not using it anywhere. 

- [x] - speed them up
- [x] - reach coverage requirement

### CODE REVIEW

**Comment by Christewart** [comment]
This also begins to partially address #678 with [deleteAppConfig](https://github.com/bitcoin-s/bitcoin-s/pull/723/files#diff-18ce41a6796a5e3386b87351d75b3c64R70)

File: chain-test/src/test/scala/org/bitcoins/chain/config/ChainAppConfigTest.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package org.bitcoins.chain.config
+
+import akka.actor.ActorSystem
+import org.bitcoins.testkit.chain.ChainUnitTest
+import org.bitcoins.testkit.util.FileUtil
+import org.scalatest.FutureOutcome
+
+class ChainAppConfigTest extends ChainUnitTest {
```
Comment (torkelrogstad) [inline chain-test/src/test/scala/org/bitcoins/chain/config/ChainAppConfigTest.scala:13]: we already have a `ChainAppConfigTest`. Merge them?

File: chain-test/src/test/scala/org/bitcoins/chain/models/BlockHeaderDAOTest.scala [hunk 2]
```diff
@@ -188,4 +188,18 @@ class BlockHeaderDAOTest extends ChainUnitTest {
...
+  it must "get an ancestor at a specified height" in {
+    blockHeaderDAO: BlockHeaderDAO =>
+      val blockHeader = BlockHeaderHelper.buildNextHeader(genesisHeaderDb)
+      val createdF = blockHeaderDAO.create(blockHeader)
+
+      val genesisF = createdF.flatMap(created =>
+        blockHeaderDAO.getAncestorAtHeight(created, 0))
+
+      genesisF.map { genesisOpt =>
+        assert(genesisOpt.get == genesisHeaderDb)
```
Comment (torkelrogstad) [inline chain-test/src/test/scala/org/bitcoins/chain/models/BlockHeaderDAOTest.scala]: nit: `genesisOpt.contains(genesisHeaderDb)`

File: chain-test/src/test/scala/org/bitcoins/chain/blockchain/ChainHandlerTest.scala [hunk 3]
```diff
@@ -258,20 +257,26 @@ class ChainHandlerTest extends ChainUnitTest {
...
+    def loop(
+        remainingHeaders: Vector[BlockHeader],
+        height: Int): Future[Assertion] = {
+      remainingHeaders match {
+        case header +: headersTail =>
+          val getHeaderF = processedHeadersF.flatMap(_.getHeader(header.hashBE))
+          val expectedBlockHeaderDb =
+            BlockHeaderDbHelper.fromBlockHeader(height, header)
+          val assertionF =
+            getHeaderF.map(tips => assert(tips.contains(expectedBlockHeaderDb)))
```
Comment (torkelrogstad) [inline chain-test/src/test/scala/org/bitcoins/chain/blockchain/ChainHandlerTest.scala]: `tips` seems like a strange name, seeing as it's `Option[BlockheaderDb]`

**Review by torkelrogstad — APPROVED** [review]

**Pull request merged**